### ISSUE TITLE
Bevy not compiling because of issues with ALSA and CPAL

### ISSUE THREAD

**Issue by anoonD** [bevyengine/bevy#6343]
## Bevy version

I also tried using version 0.8.1 but the same error came up.
```toml
[package]
name = "my_bevy_game"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
bevy = { version = "0.8.0", features = ["dynamic"] }

[profile.dev]
opt-level = 1

[profile.dev.package."*"]
opt-level = 3
```

## \[Optional\] Relevant system information

cargo/rust version:
`cargo 1.64.0 (387270bc7 2022-09-16)`

operating system:
`EndeavourOS Linux x86_64`

 kernel:
`6.0.1-zen2-1-zen`

## What you did

Hi, I'm trying to compile the [sample app in the documentation](https://bevyengine.org/learn/book/getting-started/apps/).
I also used the recommended linker `LLD` and created a `.cargo/config` file in my project directory
```toml
[target.x86_64-unknown-linux-gnu]
rustflags = [
    "-C", "link-arg=-fuse-ld=lld",
]
```
I'm pretty sure this is not an issue with the linker because this problem also comes up when I switch to `mold` or use the default linker.

## What went wrong
When I do `cargo run` it seems to be working fine until the following comes up [error.txt](https://github.com/bevyengine/bevy/files/9845439/error.txt)

At first I thought that this was an issue with dependencies so I installed the following
`sudo pacman -S libx11 pkgconf alsa-lib pipewire-alsa pulseaudio-alsa vulkan-intel`

But the error keeps coming up. I tried removing the `target` directory but the same error came up.
I'm not sure what's going on.

**Comment by dataphract** [comment]
This is due to a non-semver-compliant breaking change in the upstream ALSA dependency: see https://github.com/diwic/alsa-rs/issues/90.

**Comment by stefan-wullems** [comment]
I was able to work around this by adding `alsa = "=0.6.0"` to my dependencies. Note the `=` before the version.

**Comment by anoonD** [comment]
> I was able to work around this by adding `alsa = "=0.6.0"` to my dependencies. Note the `=` before the version.

Thanks! It's working now, when do you suspect this issue will be fixed?

**Comment by harudagondi** [comment]
> when do you suspect this issue will be fixed?

I suspect when the authors of `alsa-rs` yanks `0.6.1` and published a new `0.7.0` version instead. In the meanwhile, I think this shouldn't be closed until diwic/alsa-rs#90 is closed.
